### ISSUE TITLE
undefined method when running brew bump-formula-pr

### ISSUE THREAD

**Issue by cblecker** [Homebrew/brew#4697]
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Use bump-formula-pr to bump a formula.

- What happened (include command output)
```
$ brew bump-formula-pr --url="https://github.com/hashicorp/terraform/archive/v0.11.8.tar.gz" --sha256="c0d7a0b726579574bcfee2ae141be4e82d1c9ab4a339cc6f86f9ec38de9130fb" Formula/terraform.rb
==> Migrating cache entries...
Already up-to-date.
==> replace /https:\/\/github\.com\/hashicorp\/terraform\/archive\/v0\.11\.7\.tar\.gz/ with "https://github.com/hashicorp/terraform/archive/v0.11.8.tar.gz"
==> replace "f9a2730dcd68dad754cf0efa017d51929a7f333a89d07ddff3c6ff7b2d1e8be3" with "c0d7a0b726579574bcfee2ae141be4e82d1c9ab4a339cc6f86f9ec38de9130fb"
Error: undefined method `full_name' for nil:NilClass
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/dev-cmd/bump-formula-pr.rb:280:in `block in bump_formula_pr'
/usr/local/Homebrew/Library/Homebrew/extend/pathname.rb:278:in `block in cd'
/usr/local/Homebrew/Library/Homebrew/extend/pathname.rb:278:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/pathname.rb:278:in `cd'
/usr/local/Homebrew/Library/Homebrew/dev-cmd/bump-formula-pr.rb:264:in `bump_formula_pr'
/usr/local/Homebrew/Library/Homebrew/brew.rb:87:in `<main>'
```

- What you expected to happen
The formula would be bumped, and a PR opened.

This has actually been happening for awhile, but I hadn't had the time to file this report.

- brew doctor && brew config output
```
$ brew doctor && brew config
Your system is ready to brew.
HOMEBREW_VERSION: 1.7.1-263-g0fcf8ab
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 0fcf8abf329a7b0ef9d2f6692cde53bf1ab9748d
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 714369c7c4c64bb08e7f890b3f4b76f3d63a8940
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEVELOPER: true
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_NO_ANALYTICS_THIS_RUN: 1
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.3.7 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/bin/ruby
Clang: 9.0 build 900
Git: 2.18.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_144
macOS: 10.12.6-x86_64
CLT: 9.2.0.0.1.1510905681
Xcode: 9.2
XQuartz: 2.7.11 => /opt/X11
```

**Comment by MikeMcQuaid** [comment]
What if you instead specify `$ brew bump-formula-pr --url="https://github.com/hashicorp/terraform/archive/v0.11.8.tar.gz" --sha256="c0d7a0b726579574bcfee2ae141be4e82d1c9ab4a339cc6f86f9ec38de9130fb" terraform` instead of the formula path?

**Comment by cblecker** [comment]
Hmm okay. So I was specifying a file path because I always worked off a cloned fork location (`~/dev/homebrew-core`) rather than my system tap location (`$(brew --repo)/Library/Taps/homebrew/homebrew-core`). When I tried this without the path, it used my system tap location.

This worked, but only after I made a number of changes:
- Had to set up my git user/email on this repo location, as I use a multiple identity git setup (totally personal preference thing, and not difficult to set one time)
- Had to enable the `public_repo` scope on my `HOMEBREW_GITHUB_API_TOKEN` so it can "fork" homebrew-core.. except that I've already done this. I'd rather this token be unprivileged, as I use it mainly to get around the unauthenticated github api limits when running `brew update`. Disabling this scope afterwards breaks bump-formula-pr, as it tries blindly to refork the repo every time.
- bump-formula-pr also forces use of https cloning, rather than allowing ssh keys. If I changed `clone_url` to `ssh_url` on L288, then I was able to use SSH keys. (I have 2FA on my account, so it's easier to use the SSH key on my yubikey, rather than copy/paste an access token to authenticate).
https://github.com/Homebrew/brew/blob/a5a6da5861fdfbebdb668ab8d89b1e9bbe88ee5a/Library/Homebrew/dev-cmd/bump-formula-pr.rb#L288

I didn't quite realize how long this has been "broken" in my setup for.. I think since https://github.com/Homebrew/brew/pull/3870 merged to move away from using `hub`.

Are these items that homebrew would be open to looking at or accepting a PR on? If not, that's okay -- I'll just have to make my changes/PRs manually, rather than using bump-formula-pr anymore.

**Comment by MikeMcQuaid** [comment]
> Are these items that homebrew would be open to looking at or accepting a PR on?

Sure! I'll walk through them.

> Had to set up my git user/email on this repo location, as I use a multiple identity git setup (totally personal preference thing, and not difficult to set one time)

We'd not accept a PR for this as it's too personal.

> Disabling this scope afterwards breaks bump-formula-pr, as it tries blindly to refork the repo every time.

We'd accept a PR for this (blindly attempting to refork).

> bump-formula-pr also forces use of https cloning, rather than allowing ssh keys. If I changed clone_url to ssh_url on L288, then I was able to use SSH keys. (I have 2FA on my account, so it's easier to use the SSH key on my yubikey, rather than copy/paste an access token to authenticate).

If there's a way to customise this such that it doesn't require it being manually specified: we'd accept a PR. I also have 2FA on my account and push with a (different, keychain stored) personal access token instead.

**Comment by cblecker** [comment]
My 10 minute assessment:
> Disabling this scope afterwards breaks bump-formula-pr, as it tries blindly to refork the repo every time.

I can see why this was done. It's a simple single POST call to either fork, or get back the name of the fork.. would need to see if there is an no-scoped way of identifying if you have a fork already.

> bump-formula-pr also forces use of https cloning, rather than allowing ssh keys. If I changed clone_url to ssh_url on L288, then I was able to use SSH keys. (I have 2FA on my account, so it's easier to use the SSH key on my yubikey, rather than copy/paste an access token to authenticate).

I can see this as an option being toggled via either a git config option, or an environment variable. Is there a preference from the homebrew side?

**Comment by cblecker** [comment]
Opened #4858 with the remote protocol fix. Haven't had time to look at the "if public repo exists" check.
